pygments/lexers/func.py

```diff
@@ -0,0 +1,93 @@
+from pygments.lexer import RegexLexer, bygroups, include, words
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, \
+    Number, Whitespace, Punctuation
+
+__all__ = ['FuncLexer']
+
+
+class FuncLexer(RegexLexer):
+    name = 'FunC'
+    aliases = ['func', 'fc']
+    filenames = ['*.fc', '*.func']
+
+    identifier_allowed_symbols = r'([^;,\[\]\(\)\s~.]+)'
+    # 1. Does not start from "
+    # 2. Can start from ` and end with `, containing any character
+    # 3. Starts with underscore or { or } and have more than 1 character after it
+    # 4. Starts with letter, contains letters, numbers and underscores
+    identifier = '(?!")(`([^`]+)`|((?=_)_|(?={{){{|(?=}})}}|(?![_`{{}}])){})'.format(identifier_allowed_symbols)
```

Comment:
Since `identifier_allowed_symbols` are not used anywhere else, I suggest to join it with `identifier`.

Usually, it is a bad practice to create `regex` strings with `.format()` because of the extensive amount of `{{}}` and `\\` escaping.

Here we can totally drop it.